### PR TITLE
[SPARK-36099][CORE] Grouping exception in core/util

### PR DESCRIPTION
**Pull request by dgd-contributor**
### What changes were proposed in this pull request?
This PR group exception messages in core/src/main/scala/org/apache/spark/util

### Why are the changes needed?
It will largely help with standardization of error messages and its maintenance.

### Does this PR introduce _any_ user-facing change?
No. Error messages remain unchanged.

### How was this patch tested?
No new tests - pass all original tests to make sure it doesn't break any existing behavior.

### CODE REVIEW

**Comment by Shockang** [comment]
@dgd-contributor https://github.com/Shockang/spark/tree/SPARK-36099

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 1]
```diff
@@ -141,4 +139,161 @@ object SparkCoreErrors {
...
+
+  def taskNotSerializableError(ex: Exception): Throwable = {
+    new SparkException("Task not serializable", ex)
+  }
+
+  def returnStatementInClosureError(): Throwable = {
+    new ReturnStatementInClosureException
+  }
+
+  def cannotResolvesAmbiguouslyBaseError(base: String, resolved: String): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:201]: cannotResolveToMultipleFilesError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 2]
```diff
@@ -141,4 +139,161 @@ object SparkCoreErrors {
...
+    new SparkException(s"$base resolves ambiguously to multiple files: $resolved")
+  }
+
+  def keyMustNotBeNullError(): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:205]: keyIsNullError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 3]
```diff
@@ -141,4 +139,161 @@ object SparkCoreErrors {
...
+
+  def endOfStreamError(): Throwable = {
+    new NoSuchElementException("End of stream")
+  }
+
+  def endOfFileError(): Throwable = {
+    new EOFException("End of file before fully reading buffer")
+  }
+
+  def couldNotWriteBufferToOutputSteamError(): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:221]: cannotWriteBufferToOutputSteamError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 4]
```diff
@@ -141,4 +139,161 @@ object SparkCoreErrors {
...
+
+  def executorAlreadyShutdownError(): Throwable = {
+    new RejectedExecutionException("Executor already shutdown")
+  }
+
+  def exceptionThrownInAwaitResultError(t: Throwable): Throwable = {
+    new SparkException("Exception thrown in awaitResult: ", t)
+  }
+
+  def failedToCreateTempDirectoryError(root: String, maxAttempts: Int): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:233]: failToCreateTempDirectoryError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 5]
```diff
@@ -141,4 +139,161 @@ object SparkCoreErrors {
...
+    new IOException("Failed to create a temp directory (under " + root + ") after " +
+      maxAttempts + " attempts!")
+  }
+
+  def failedToDeleteFileError(
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:238]: failToDeleteFileError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 6]
```diff
@@ -141,4 +139,161 @@ object SparkCoreErrors {
...
+
+  def failToGetTempDirectoryError(dir: String): Throwable = {
+    new IOException(s"Failed to get a temp directory under [$dir].")
+  }
+
+  def yarnLocalDirsCannotBeEmptyError(): Throwable = {
+    new Exception("Yarn Local dirs can't be empty")
+  }
+
+  def processExistedError(command: Seq[String], exitCode: Int): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:261]: processExitedError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 7]
```diff
@@ -141,4 +139,161 @@ object SparkCoreErrors {
...
+
+  def destinationMustBeRelativeError(): Throwable = {
+    new IOException("Destination must be relative")
+  }
+
+  def failToLoadSparkPropertiesError(filename: String, e: Throwable): Throwable = {
+    new SparkException(s"Failed when loading Spark properties from $filename", e)
+  }
+
+  def failedToStartServiceOnPortError(serviceString: String, startPort: Int): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:281]: failToStartServiceOnPortError

**Review by allisonwang-db — COMMENTED** [review]
Left a few comments. Please also resolve the merge conflicts.

**Comment by dgd-contributor** [comment]
As we and @HyukjinKwon discussed contributing by sharing accounts, we will close this PR, Please take a look at this PR https://github.com/apache/spark/pull/34188

**Pull request closed without merging**